lib/overcommit/hook/pre_push/protected_branches.rb

```diff
@@ -13,6 +14,15 @@ def run
 
     private
 
+    def log_pushed_refs
+      log_msg = "\nPushed refs:\n\t#{pushed_refs_str}"
+      Overcommit::Utils.log.debug(log_msg)
```

Comment:
Printing debug messages breaks the output of the hook with the way we currently output `OK`/`FAIL` at the end of the hook's run, so I'd be hesitant to commit this in even if it only appeared with `OVERCOMMIT_DEBUG=1`.

This has come up a few times, so it seems like we need to expose a call that allows a hook to aggregate log messages which are printed in addition to its regular output when debugging is enabled.